sagemaker-pipeline-compare-model-versions/notebook.ipynb

```diff
@@ -0,0 +1,703 @@
+    "\n",
+    "#### SageMaker Pipeline steps\n",
+    "SageMaker pipelines works on the concept of steps. The order steps are executed in is inferred from the dependencies each step have. If a step has a dependency on the output from a previous step, it's not executed until after that step has completed successfully. \n",
+    "\n",
+    "Currently, there are 6 different types of steps supported.\n",
+    "\n",
+    "| Step | Description |\n",
+    "| ----------- | ----------- |\n",
+    "| Processing | Runs a processing job using a provided script |\n",
+    "| Training | Runs a training job using the container of your choice |\n",
+    "| Tuning | Runs a hyperparameter tuning job |\n",
+    "| Transform | Runs a batch-transform job to either process data or perform batch-inference |\n",
+    "| CreateModel | Creates a model to be used with a Transform step |\n",
+    "| RegisterModel | Registers a new model or registers a new model version with the Model Registry |\n",
+    "| Condition | Condition step to branch the execution depending on some condition |\n",
+    "\n",
+    "#### SageMaker Pipeline concepts\n",
+    "\n",
+    "##### Pipeline DAG\n",
+    "When creating a SageMaker Pipeline, SageMaker creates a Direct Acyclic Graph, DAG, that customers can visualize in Amazon SageMaker Studio. The DAG can be used to track pipeline executions, outputs and metrics. In this notebook, a SageMaker pipeline with the following DAG is created:\n",
+    "\n",
+    "![Customer churn training pipeline](images/pipeline.png \"Customer churn training pipeline\")\n",
+    "\n",
+    "\n",
+    "##### Parameters\n",
+    "SageMaker Pipeline Parameters are input parameters specified when triggering a pipeline execution. They need to explicitly defined when creating the pipeline and contain default values. Currently, three types of parameters are supported:\n",
+    "* String parameters\n",
+    "* Integer parameters\n",
+    "* Float parameters\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Predict customer churn with XGboost\n",
+    "\n",
+    "### Data\n",
+    "\n",
+    "This notebook uses a synthetic dataset.\n",
+    "\n",
+    "This dataset contains information about phone carrier customers, such as customer location, phone number, number of phone calls etc.\n",
+    "\n",
+    "\n",
+    "### Overview \n",
+    "The purpose of this notebook is to demonstrate how SageMaker Pipelines can be used to preprocess, train, evaluate and push new machine learning models into the SageMaker model registry for version comparison. All scripts to preprocess the data and evaluate the trained model have been prepared in advance and are available here: \n",
```

Comment:
please add a disclaimer at the beginning that this notebook recommends using SageMaker Studio as reader will not get the UI bits and DAG etc on notebook instances.